src/openrpc/media_access.json

```diff
@@ -0,0 +1,580 @@
+                        "xrn:firebolt:capability:media-access:images"
+                    ]
+                }
+            ],
+            "summary": "Access available image MediaFiles.",
+            "params": [
+                {
+                    "name": "volumes",
+                    "summary": "A set of queries used to match which volumes will be matched.",
+                    "required": false,
+                    "schema": {
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                },
+                {
+                    "name": "files",
+                    "summary": "A set of queries used to match which files will be matched.",
+                    "required": false,
+                    "schema": {
+                        "$ref": "#/components/schemas/MediaFile"
+                    }
+                }
+            ],
+            "result": {
+                "name": "files",
+                "summary": "A set of MediaFiles that matched the query.",
+                "required": true,
+                "schema": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/components/schemas/MediaFile"
+                    }
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default example #1",
+                    "params": [
+                        {
+                            "name": "volumes",
+                            "value": {
+                                "type": "all"
+                            }
+                        }
+                    ],
+                    "result": {
+                        "name": "files",
+                        "value": [
+                            {
+                                "uri": "/usb/my-usb-drive/photo1.jpg",
+                                "volume": {
+                                    "uri": "/usb/my-usb-drive",
+                                    "name": "My USB Drive",
+                                    "type": "usb"
+                                },
+                                "type": "image/jpg"
+                            },
+                            {
+                                "uri": "/local/photos/photo1.jpg",
+                                "volume": {
+                                    "uri": "/local/photos",
+                                    "name": "photos",
+                                    "type": "local"
+                                    },
+                                "type": "image/jpg"
+                            }
+                        ]
+                    }
+                }
+            ]
+        },
+        {
+            "name": "media",
+            "tags": [
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:media-access:*"
```

Comment:
@kpears201 : How should we denote that a method requires one of many possible capabilities?
Currently, if we put more than one capability here, the method requires ALL of them.

I'm really hoping to avoid:

```
"x-uses": {
  "oneOf": [
  ]
}
```

as that would fork all of our parsing logic...

I guess we could do something like:

```
"x-uses": [
],
"x-uses-operator": "or"
```

Reply:
Is it necessary to check the capability here?

Consider these two cases:

App has permission to `xrn:firebolt:capability:media-access:audio` but does not have permission to `xrn:firebolt:capability:media-access:video` (or any other media-access caps)

The app calls MediaAccess.media("all").
What will it return?
Won't it just filter out the videos and not return them and only return the audio.
Now consider removing `xrn:firebolt:capability:media-access:audio` permission from the app.

Seems unnecessary that this now returns an error because the app doesn't have permission to anything. Can't we just return an empty list of files because we have filtered out all files it does not have permission to?

Reply:
that might work, but we have historically tried to avoid api implementations checking specific capabilities, and doing that at the front door. I suppose at some point it is inevitable, though.